source/github-helpers/toast.tsx

```diff
@@ -12,14 +12,14 @@ export function ToastSpinner(): JSX.Element {
 }
 
 type ProgressCallback = (message: string) => void;
-type Task = (progress?: ProgressCallback) => Promise<unknown>;
-export default async function showToast<TTask extends Task>(
-	task: TTask,
+type Task<T> = (progress?: ProgressCallback) => Promise<T>;
+export default async function showToast<TaskOrError extends Task<any> | Error>(
+	task: TaskOrError,
 	{
 		message = 'Bulk actions currently being processed.',
 		doneMessage = 'Bulk action processing complete.',
 	} = {},
-): Promise<ReturnType<TTask>> {
+): Promise<[TaskOrError] extends [Task<infer R>] ? R : void> {
```

Comment:
Since we never use the return value of `showToast`, all those types could be simplified.